### ISSUE TITLE
Publisher and Publisher Place (for journalArticle)

### ISSUE THREAD

**Issue by rmzelle** [citation-style-language/zotero-bits#16]
It seems to be customary in some countries to include the publisher and publisher-place to cited journal articles (see link below). This can be supported by adding "Publisher"/**publisher** and "Publisher Place"/**publisherPlace** to _journalArticle_ (mapped to **publisher** and **publisher-place** in CSL).

http://forums.zotero.org/discussion/8186

**Comment by nitsnatsnok** [comment]
Please add support for citing publisher and publisher-place for journal-articles! It’s common to cite it in german, at least in linguistics. Workarounds are really disturbing. Thank you!

**Comment by adam3smith** [comment]
I don't think there's disagreement on this.

**Comment by wbt** [comment]
What are the primary hold-ups then?  I get the impression I'm significantly underestimating the challenges involved in adding these fields.

**Comment by adam3smith** [comment]
As we explain in various location, the hold-up is entirely Zotero's sync infrastructure. Actually adding the fields in the Zotero clients takes <5mins, but we can't make any changes until Zotero is completely switched to an entirely different method of syncing. Hope is that that's going to be pretty soon, but in all honesty we don't know. That was the hope a year ago, too, so impossible to say.

**Comment by artorius** [comment]
Any update on this?  I can manually enter the publisher in my bibliography, it would just be nice for Zotero to do this automagically.

**Comment by rmzelle** [comment]
The only update is that Zotero 5.0 (with the new sync infrastructure necessary to start changing Zotero fields) has just [entered beta](https://forums.zotero.org/discussion/59829/zotero-50-beta/), but it might still be a while before these changes are implemented. We'll advocate for this change to be made when we see some movement by the Zotero developers, but it's still out of our hands for now (when I say we, I mean Zotero users and CSL developers).

In the meantime, you can use the workaround of storing the publisher information in Zotero's "Extra" field in the following format:

```
{:publisher: Oxford University Press}
```

or

```
{:publisher: Galleria Bonino}
{:publisher-place: New York}
```

(this should work for any CSL variable you'd like to add to a given item type)

**Comment by jrochkind** [comment]
@rmzelle I see Zotero 5.0 was released a few months ago, does that effect the blocking concerns for this issue?

**Comment by adam3smith** [comment]
No, the blocking concern is for all 4.0 clients to be cut off from sync and the release of Zotero 5.1

**Comment by jrochkind** [comment]
ok, thanks. any expected timeframe for that to happen?

**Comment by adam3smith** [comment]
Very likely this year -- the 5.x update also means that future changes are easy to make, so we don't need to cram every desired change into the first release. E.g. adding this to Zotero will break something like 70-90% of CSL styles (which will then erroneously print that information for journals), so I'm not sure if we'll do this to us during a busy time.

**Comment by niranjantdesai** [comment]
Any update on this?

**Comment by niranjantdesai** [comment]
> The only update is that Zotero 5.0 (with the new sync infrastructure necessary to start changing Zotero fields) has just [entered beta](https://forums.zotero.org/discussion/59829/zotero-50-beta/), but it might still be a while before these changes are implemented. We'll advocate for this change to be made when we see some movement by the Zotero developers, but it's still out of our hands for now (when I say we, I mean Zotero users and CSL developers).
> 
> In the meantime, you can use the workaround of storing the publisher information in Zotero's "Extra" field in the following format:
> 
> ```
> {:publisher: Oxford University Press}
> ```
> or
> 
> ```
> {:publisher: Galleria Bonino}
> {:publisher-place: New York}
> ```
> (this should work for any CSL variable you'd like to add to a given item type, but it's a bit of a hack)

I'm using the [Better Bib(La)Tex](https://github.com/retorquere/zotero-better-bibtex) add-on and I'm unable to get this hack working with it.

**Comment by Ciantic** [comment]
Hard to keep track of these. 5.0 is obviously released, but what is the issue we should follow for the new fields system?

**Comment by craymichael** [comment]
> > The only update is that...
> > In the meantime, you can use the workaround of storing the publisher information in Zotero's "Extra" field in the following format:
> > ```
> > {:publisher: Oxford University Press}
> > ```
> > ...
> > (this should work for any CSL variable you'd like to add to a given item type, but it's a bit of a hack)
> 
> I'm using the [Better Bib(La)Tex](https://github.com/retorquere/zotero-better-bibtex) add-on and I'm unable to get this hack working with it.

I found this did not work for me either. If this doesn't work for you, you will want to do something like this in the "Extra" field:

```
bibtex[publisher=Oxford University Press]
```

Multiple fields can be added using the following: 

```
bibtex[publisher=Oxford University Press;origdate=1856]
```

See [the documentation](https://retorque.re/zotero-better-bibtex/customized-exports/) for further details.

**Comment by adam3smith** [comment]
@Ciantic there's no single ticket for new fields at this time

@craymichael no, that's incorrect, read further down in the docs. The non-bibtex syntax is actually preferable for most people as it will work both for citations generated from Zotero _and_ for bibtex export. That said, you can leave out the curly brackets and colons now. `publisher: Oxford University Press` works just as well and is less messy.

If these don't get exported by BetterBibLaTeX, you should ask for help that on the project's issue tracker. The developer is very responsive.

**Comment by craymichael** [comment]
@adam3smith Thanks for pointing that out. However, the CSL style did not work for me nor did it for @niranjantdesai - I have edited my response to reflect this. Additionally, the `publisher: Oxford University Press` syntax does not work for me either.

**Comment by adam3smith** [comment]
Right, so the syntax works (try with `original-date` exporting to BibLaTeX, for example) but only if exporting the Zotero field / CSL variable to BibTeX/BibLaTeX is implemented in BBT, which doesn't appear to be the case for publishers in journal articles. I'd assume the developer would be willing to implement that if you ask. Same for other fields as long as they have clear Bib(la)tex mappings.

**Comment by ronanpaixao** [comment]
Any updates on the addition of a "Place" field for periodicals and magazines? When implementing this, can anyone also include DOI field for magazines?

**Comment by bwiernik** [comment]
You can enter all of these in Extra as needed. They will likely be added in a future version
